### PR TITLE
Adds Fulpstation spell granters and associated space ruins

### PR DESCRIPTION
**Pull request by QuiteLiterallyAnything**
## About The Pull Request
This PR adds in two new spell granters for both Fulpstation spells currently available. 
<details><summary>Expand to see the sprites for the new spell granters</summary>

image
_Both icons are slightly animated._

</details>

This PR also adds in two new space ruins containing both spell granters as loot. (The ruin for the cateor spell granter has a mild hazard to account for the spell's value in combat.)
<details><summary>Expand to see direct StrongDMM screenshots of both new ruins</summary>

image
_"Direct Cat Meteor" spell book ruin._

image
_"Summon Dancefloor" spell book ruin._
</details>

## Why It's Good For The Game
There's practically no way for players to get either of these spells at the moment (aside from hoping for extreme luck with the Greed ruin on Lavaland,) and this PR addresses that by adding in a way to get them.
## Changelog
:cl:
add: Added spell book granters for both "Summon Dancefloor" and "Direct Cat Meteor"
add: Added two new space ruins containing the previously mentioned spell book granters
/:cl:

### CODE REVIEW

**Review by JohnFulpWillard — APPROVED** [review]
now if only we had pop for wiz rounds

File: fulp_modules/features/spells/spell_granters/summon_dancefloor.dm [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+/obj/item/book/granter/action/spell/fulp/summon_dancefloor/recoil(mob/living/user)
+	. = ..()
+	to_chat(user, span_warning("You feel compelled to try out an epic dance move!"))
+	user.Immobilize(6 SECONDS)
+	playsound(get_turf(user), 'sound/magic/blind.ogg', 37, frequency = -1)
+	user.emote("flip")
+	user.emote("spin")
+	user.emote("snap")
+	sleep(4 SECONDS)
```
Comment (JohnFulpWillard) [inline fulp_modules/features/spells/spell_granters/summon_dancefloor.dm:23]: Actually please get rid of sleeps, use timers instead

**Review by JohnFulpWillard — CHANGES_REQUESTED** [review]

File: fulp_modules/features/spells/spell_granters/summon_dancefloor.dm (hunk 1 above)
Comment (QuiteLiterallyAnything) [reply fulp_modules/features/spells/spell_granters/summon_dancefloor.dm]: Done, thanks for the review! (ﾉ◕ヮ◕)ﾉ*:･ﾟ✧

File: fulp_modules/features/spells/spell_granters/summon_dancefloor.dm [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	playsound(get_turf(user), 'sound/effects/magic/blind.ogg', 37, frequency = -1)
+	user.emote("flip")
+	user.emote("spin")
+	user.emote("snap")
+	addtimer(CALLBACK(src, PROC_REF(comedic_premonition), user), 4 SECONDS)
+	addtimer(CALLBACK(src, PROC_REF(dance_injury), user), 6 SECONDS)
+
+/obj/item/book/granter/action/spell/fulp/summon_dancefloor/proc/comedic_premonition(mob/living/user)
+	to_chat(user, span_bolddanger("You might've just broken something..."))
+	playsound(get_turf(user), 'sound/effects/wounds/crack1.ogg', 37, frequency = -1)
```
Comment (JohnFulpWillard) [inline fulp_modules/features/spells/spell_granters/summon_dancefloor.dm:28]:
```suggestion

/obj/item/book/granter/action/spell/fulp/summon_dancefloor/proc/comedic_premonition(mob/living/user)
	to_chat(user, span_bolddanger("You might've just broken something..."))
	playsound(get_turf(user), 'sound/effects/wounds/crack1.ogg', 37, frequency = -1)
	addtimer(CALLBACK(src, PROC_REF(dance_injury), user), 2 SECONDS)
```

**Pull request merged**